main.go

```diff
@@ -38,7 +144,175 @@ func httpHandler(w http.ResponseWriter, r *http.Request) {
+	}
+	analysisJob := jobs["analysis"]
+	if analysisJob == nil {
+		return false
+	}
+
+	// Verify that there is no container or services.
+	if analysisJob.Container != nil || len(analysisJob.Services) > 0 {
+		return false
+	}
+
+	// Verify that the workflow runs on ubuntu-latest.
+	if len(analysisJob.RunsOn.Labels) != 1 || analysisJob.RunsOn.Labels[0].Value != "ubuntu-latest" {
+		return false
+	}
+
+	// Verify that there are no env vars set.
+	if analysisJob.Env != nil {
+		return false
+	}
+
+	// Get steps in job.
+	steps := analysisJob.Steps
+
+	if steps == nil || len(steps) > 4 {
+		return false
+	}
+
+	// Verify that steps are valid (checkout, scorecard-action, upload-artifact, upload-sarif).
+	for _, step := range steps {
+		stepName := step.Exec.(*actionlint.ExecAction).Uses.Value
+		stepName = stepName[:strings.Index(stepName, "@")] // get rid of commit sha.
+
+		switch stepName {
+		case
+			"actions/checkout",
+			"ossf/scorecard-action",
+			"actions/upload-artifact",
+			"github/codeql-action/upload-sarif":
+		default:
+			return false
+		}
+	}
+
+	return true
+}
+
+// Source: https://github.com/sigstore/cosign/blob/18d2ce0b458018951f7356db911467a427a8dffe/pkg/cosign/tlog.go#L247
+func verifyTLogEntry(ctx context.Context, rekorClient *client.Rekor, uuid string) (*models.LogEntryAnon, error) {
+	params := entries.NewGetLogEntryByUUIDParamsWithContext(ctx)
+	params.EntryUUID = uuid
+
+	lep, err := rekorClient.Entries.GetLogEntryByUUID(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(lep.Payload) != 1 {
+		return nil, errors.New("UUID value can not be extracted")
+	}
+	e := lep.Payload[params.EntryUUID]
+	if e.Verification == nil || e.Verification.InclusionProof == nil {
+		return nil, errors.New("inclusion proof not provided")
+	}
+
+	hashes := [][]byte{}
+	for _, h := range e.Verification.InclusionProof.Hashes {
+		hb, _ := hex.DecodeString(h)
+		hashes = append(hashes, hb)
+	}
+
+	rootHash, _ := hex.DecodeString(*e.Verification.InclusionProof.RootHash)
+	leafHash, _ := hex.DecodeString(params.EntryUUID)
+
+	v := logverifier.New(rfc6962.DefaultHasher)
+	if err := v.VerifyInclusionProof(*e.Verification.InclusionProof.LogIndex, *e.Verification.InclusionProof.TreeSize, hashes, rootHash, leafHash); err != nil {
+		return nil, err
+	}
+
+	// Verify rekor's signature over the SET.
+	payload := bundle.RekorPayload{
+		Body:           e.Body,
+		IntegratedTime: *e.IntegratedTime,
+		LogIndex:       *e.LogIndex,
+		LogID:          *e.LogID,
+	}
+
+	rekorPubKeys, err := cosign.GetRekorPubs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var entryVerError error
+	for _, pubKey := range rekorPubKeys {
+		entryVerError = cosign.VerifySET(payload, []byte(e.Verification.SignedEntryTimestamp), pubKey.PubKey)
+		// Return once the SET is verified successfully.
+		if entryVerError == nil {
+			if pubKey.Status != tuf.Active {
+				fmt.Fprintf(os.Stderr, "**Info** Successfully verified Rekor entry using an expired verification key\n")
```

Comment:
should this return an error? Mhh probably not, if it was a rotated key. How about revoked/compromised keys?

cc @asraa